ios/browser/api/web_image/image_downloader.mm

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#import "brave/ios/browser/api/web_image/image_downloader.h"
+#import "brave/ios/browser/svg/svg_image.h"
+
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "skia/ext/skia_utils_ios.h"
+#include "third_party/skia/include/core/SkBitmap.h"
+#include "ui/gfx/image/image.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+namespace brave {
+ImageDownloader::ImageDownloader(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : image_fetcher_(url_loader_factory) {}
+
+ImageDownloader::~ImageDownloader() {}
+
+int ImageDownloader::DownloadImage(const GURL& url,
+                                   int max_image_size,
+                                   ImageDownloadCallback callback) {
+  static int downloaded_image_count = 0;
+  int local_download_id = ++downloaded_image_count;
+
+  GURL local_url(url);
+  __block ImageDownloadCallback local_callback = std::move(callback);
+
+  image_fetcher::ImageDataFetcherBlock ios_callback =
+      ^(NSData* data, const image_fetcher::RequestMetadata& metadata) {
+        if (metadata.http_response_code ==
+            image_fetcher::RequestMetadata::RESPONSE_CODE_INVALID) {
+          return;
```

Comment:
You're not calling `local_callback` here and probably should anyways (consider when we would call this with swift concurrency, it would never continue the task)

Reply:
Fixed. I was doing what Chromium does. It doesn't call the callback :o